tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * This script is written for a one time use only to facilitate the normalization of the consent feature of candidate parameters. 
+ * 
+ * This script imports existing consent data from `participant_status` and `consent_info_history` into new normalized tables.
+ * 
+ * A SQL file is exported to path: loris_root/SQL/Cleanup_patches/delete_old_consent_tables.sql
+ * after running this script. Source this file to delete the now deprecated tables and columns.
+ * 
+ * affected tables:
+ * - participant_status
+ * - consent_info_history
+ * - consent
+ * - candidate_consent_rel
+ * - candidate_consent_history
+ * 
+ * PHP Version 7
+ * 
+ * @category Main 
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+
+$config = NDB_Config::singleton();
+$db     =& Database::singleton();
+
+// Get consent details from Config.xml
+$consentConfig = $config->getSetting('ConsentModule');
+$useConsent = $consentConfig['useConsent'];
+$consents = $consentConfig['Consent'];
+
+// Format array of consents
+foreach($consents as $consent) {
+    $consentName = $consent['name'];
+    $consentLabel = $consent['label'];
+    $consentList[$consentName] = $consentLabel;
+}
+///////////////////////////////////////
+// VALIDATE EXISTING CONSENT DETAILS //
+///////////////////////////////////////
+
+$errors = [];
+// Check for when consent columns exist in participant_status but not in Config.xml
+$columnQuery = "SELECT Column_name FROM Information_schema.columns WHERE
+                Column_name LIKE '%consent%' AND Table_name LIKE 'participant_status'";
```

Comment:
from the projects that i have seen, yes, every column in `participant_status` for a consent has the word "consent" in it to describe what it is. checks both ways (if consent in config.xml, is it in table? if consent in table, is it in config.xml?) are looked at in the import script because what if a consent is, for what ever reason, deleted from config.xml but still exists in `participant_status`? we would want someone to look into that. the bulk of the import is done in a for loop for each consent in config.xml after the data validation though